proto/regen/ecocredit/v1alpha1/genesis.proto

```diff
@@ -0,0 +1,80 @@
+syntax = "proto3";
+package regen.ecocredit.v1alpha1;
+
+import "regen/ecocredit/v1alpha1/types.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/regen-network/regen-ledger/x/ecocredit";
+
+// GenesisState defines ecocredit module's genesis state.
+message GenesisState {
+  // Params contains the updateable global parameters for use with the x/params
+  // module
+  Params params = 1 [ (gogoproto.nullable) = false ];
+
+  // class_info is the list of credit class info.
+  repeated ClassInfo class_info = 2;
+
+  // batch_info is the list of credit batch info.
+  repeated BatchInfo batch_info = 3;
+
+  // id_seq is used to get next class/batch id.
+  uint64 id_seq = 4;
+
+  // balances is the list of credit batch tradable/retired units.
+  repeated Balance balances = 5;
+
+  // supplies is the list of credit batch tradable/retired supply.
+  repeated Supply supplies = 6;
+
+  // precisions is the list of decimal precision of a credit batch.
+  repeated Precision precisions = 7;
+}
+
+// Precision represents a credit batch precision with a batch_denom and
+// max_decimal_places.
+message Precision {
+  // batch_denom is the unique ID of the credit batch.
+  string batch_denom = 1;
+
+  // max_decimal_places is the new maximum number of decimal places that can be
+  // used to represent some quantity of credit units. It is an experimental
+  // feature to concretely explore an idea proposed in
+  // https://github.com/cosmos/cosmos-sdk/issues/7113.
+  uint32 max_decimal_places = 2;
+}
+
+// Balance represents tradable or retired units of a credit batch with an
+// account address, batch_denom, and balance.
+message Balance {
+  // address is the account address of the account holding credits.
+  string address = 1;
+
+  // batch_denom is the unique ID of the credit batch.
+  string batch_denom = 2;
+
+  // balance is the tradable or retired balance of the credit batch.
+  string balance = 3;
+
+  // type is the credit batch balance type.
+  Type type = 4;
+
+  // Type is the credit balance type.
+  enum Type {
+    // TYPE_UNSPECIFIED is the unknown credit balance type.
+    TYPE_UNSPECIFIED = 0;
+    // TYPE_TRADABLE is the tradable balance type.
+    TYPE_TRADABLE = 1;
+    // TYPE_RETIRED is the retired balance type.
+    TYPE_RETIRED = 2;
+  }
+}
+
+// Supply represents a tradable or retired supply of a credit batch.
+message Supply {
+  // batch_denom is the unique ID of the credit batch.
+  string batch_denom = 1;
+
+  // supply is the tradable or retired supply of the credit batch.
+  string supply = 2;
+}
```

Comment:
Can you update this to the same format as above (e.g. with separate `tradable_balance` and `retired_balance` fields in the `Supply` message?)